lib/utils.js

```diff
@@ -60,12 +58,21 @@ exports.domEach = function(cheerio, fn) {
  * string and then parsing the resultant markup.
  *
  * @argument {Object} dom - The htmlparser2-compliant DOM structure
- * @argument {Object} options - The parsing/rendering options
  */
-exports.cloneDom = function(dom, options) {
-  options = assign({}, options, { _useHtmlParser2: true });
+exports.cloneDom = function(dom) {
+  function filterOutParent(node) {
+    if (
+      "length" in dom
+        ? Array.prototype.some.call(dom, function(el) {
+          return node === el.parent;
+        })
+        : node === dom.parent
+    ) {
+      return null;
+    }
+  }
 
-  return parse(render(dom, options), options, false).children;
+  return cloneDeepWith(dom, filterOutParent);
```

Comment:
The documentation still describes the heuristic you're replacing. Could you update that,too?